packages/web/src/components/cells/code.tsx

```diff
@@ -253,66 +253,64 @@ export default function CodeCell(props: {
           )}
         </DialogContent>
       </Dialog>
-
-      <div className="relative group/cell" id={`cell-${props.cell.id}`}>
-        <div
-          className={cn(
-            'border rounded-md group',
-            cell.status === 'running' && 'ring-1 ring-run-ring border-run-ring',
-            (cellMode === 'generating' || cellMode === 'fixing') &&
-              'ring-1 ring-ai-ring border-ai-ring',
-            cell.status !== 'running' &&
-              cellMode !== 'generating' &&
-              cellMode !== 'fixing' &&
-              'focus-within:ring-1 focus-within:ring-ring focus-within:border-ring',
-          )}
-        >
-          <Header
-            cell={cell}
-            runCell={runCell}
-            stopCell={stopCell}
-            onDeleteCell={onDeleteCell}
-            generate={generate}
-            cellMode={cellMode}
-            setCellMode={setCellMode}
-            prompt={prompt}
-            setPrompt={setPrompt}
-            updateFilename={updateFilename}
-            filenameError={filenameError}
-            setFilenameError={setFilenameError}
-            fullscreen={fullscreen}
-            setFullscreen={setFullscreen}
-            setShowStdio={setShowStdio}
-            onAccept={onAcceptDiff}
-            onRevert={onRevertDiff}
-          />
-
-          {cellMode === 'reviewing' ? (
-            <DiffEditor original={cell.source} modified={newSource} />
-          ) : (
-            <>
-              <div className={cn(cellMode !== 'off' && 'opacity-50')}>
-                <CodeEditor
-                  cell={cell}
-                  runCell={runCell}
-                  updateCellOnServer={updateCellOnServer}
-                  readOnly={['generating', 'prompting'].includes(cellMode)}
-                />
-              </div>
-              <CellOutput
-                cell={cell}
-                show={showStdio}
-                setShow={setShowStdio}
-                fixDiagnostics={aiFixDiagnostics}
-                cellMode={cellMode}
-                fullscreen={fullscreen}
-                setFullscreen={setFullscreen}
-              />
-            </>
-          )}
-        </div>
+      <div
+        className={cn(
```

Comment:
I haven't tried to refactor this yet but I suspect we can improve states here as well.